pyccel/codegen/printing/ccode.py

```diff
@@ -551,6 +551,10 @@ def _print_PythonBool(self, expr):
         return '({} != 0)'.format(value)
 
     def _print_Literal(self, expr):
+        if isinstance(expr, LiteralInteger) and get_final_precision(expr) == 8:
+            return f"{repr(expr.python_value)}l"
+        elif isinstance(expr, LiteralFloat) and get_final_precision(expr) == 4:
+            return f"{repr(expr.python_value)}l"
```

Comment:
I think this could be simply written like:
```suggestion
        if isinstance(expr, (LiteralInteger, LiteralFloat)) and get_final_precision(expr) in (8, 4):
            return f"{repr(expr.python_value)}l"
```

Reply:
In the case of `LiteralFloat`, why `get_final_precision` should return `4` and not `8` like for `LiteralInteger`?

Reply:
If it's not the case, I think you should replace `l` with `f` in line 557.

Reply:
yeah after I change from string formatting to f-string I didn't pay attention to the 'f'
I don't know if there is a way to explicitly get a float (kind=4) instead of the default float(kind=8)
```python
def fn(a: 'float32'):
    pass

fn(2.3)
```
the example above will not work and I wonder if there is a way to make it work with just literal

Reply:
`2.3` has type `float`, for `float32` you need to do:
```python
import numpy as np

def fn(a: 'float32'):
    pass

fn(np.float32(2.3))
```

Reply:
> I think this could be simply written like:
> 
> ```python
> if isinstance(expr, (LiteralInteger, LiteralFloat)) and get_final_precision(expr) in (8, 4):
>     return f"{repr(expr.python_value)}l"
> ```

I disagree. This doesn't do the same thing as the code above. It is equivalent to:
```python
if isinstance(expr, LiteralInteger) and get_final_precision(expr) in (4,8):
    return f"{repr(expr.python_value)}l"
elif isinstance(expr, LiteralFloat) and get_final_precision(expr) in (4,8):
    return f"{repr(expr.python_value)}l"
```
I don't think an `int32` should have an `l`